plugins/utils/neoclip.nix

```diff
@@ -0,0 +1,243 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "neoclip";
+  originalName = "nvim-neoclip.lua";
+  defaultPackage = pkgs.vimPlugins.nvim-neoclip-lua;
+
+  maintainers = [ maintainers.GaetanLepage ];
+
+  settingsOptions = {
+    history = helpers.defaultNullOpts.mkUnsignedInt 1000 ''
+      The max number of entries to store.
+    '';
+
+    enable_persistent_history = helpers.defaultNullOpts.mkBool false ''
+      If set to `true` the history is stored on `VimLeavePre` using `sqlite.lua` and lazy loaded when
+      querying.
+    '';
+
+    length_limit = helpers.defaultNullOpts.mkUnsignedInt 1048576 ''
+      The max number of characters of an entry to be stored (default 1MiB).
+      If the length of the yanked string is larger than the limit, it will not be stored.
+    '';
+
+    continuous_sync = helpers.defaultNullOpts.mkBool false ''
+      If set to `true`, the runtime history is synced with the persistent storage everytime it's
+      changed or queried.
+
+      If you often use multiple sessions in parallel and wants the history synced you might want to
+      enable this.
+
+      Off by default; it can cause delays because the history file is written everytime you yank
+      something.
+      Although, in practice no real slowdowns were noticed.
+
+      Alternatively see `db_pull` and `db_push` functions in the
+      [README](https://github.com/AckslD/nvim-neoclip.lua#custom-actions).
+    '';
+
+    db_path =
+      helpers.defaultNullOpts.mkStr { __raw = "vim.fn.stdpath('data') .. '/databases/neoclip.sqlite3'"; }
+        ''
+          The path to the sqlite database to store history if `enable_persistent_history=true`.
+          Defaults to `$XDG_DATA_HOME/nvim/databases/neoclip.sqlite3`.
+        '';
+
+    filter = helpers.defaultNullOpts.mkLuaFn null ''
+      A function to filter what entries to store (default all are stored).
+      This function filter should return `true` (include the yanked entry) or `false` (don't include
+      it) based on a table as the only argument, which has the following keys:
+
+      - `event`: The event from `TextYankPost` (see ``:help TextYankPost` for which keys it contains).
+      - `filetype`: The filetype of the buffer where the yank happened.
+      - `buffer_name`: The name of the buffer where the yank happened.
+    '';
+
+    preview = helpers.defaultNullOpts.mkBool true ''
+      Whether to show a preview (default) of the current entry or not.
+      Useful for for example multiline yanks.
+      When yanking the filetype is recorded in order to enable correct syntax highlighting in the
+      preview.
+
+      NOTE: in order to use the dynamic title showing the type of content and number of lines you
+      need to configure `telescope` with the `dynamic_preview_title = true` option.
```

Comment:
![image](https://github.com/nix-community/nixvim/assets/33058747/1c41400f-97fb-45d5-bb41-273bb297a1ae)

Reply:
Found this: https://crates.io/crates/mdbook-alerts

Let's open a separate issue and we can investigate whenever there's time/interest.

Reply:
There's also this: https://rust-lang.github.io/mdBook/format/mdbook.html

```html
<div class="warning">

This is a bad thing that you should pay attention to.

Warning blocks should be used sparingly in documentation, to avoid "warning
fatigue," where people are trained to ignore them because they usually don't
matter for what they're doing.

</div>
```

Reply:
#1791